source/blueprints/banking.rst

```diff
@@ -0,0 +1,546 @@
+  - ``default_eval`` - (bool, optional): Use system validation (default: True)
+  - ``hints`` - (list, optional): Developer defined list of keywords to verify theuser input against
+  - ``custom_eval`` - (func, optional): custom validation function (should return bool:validated or not)
+
+A simple example of slot filling logic is shown below:
+
+.. code:: python
+    
+    #slot filling logic requires a form which has your needed entities for the intent 
+    balance_form = {
+        'entities':[
+        FormEntity(
+	    #specify the entity custom or system 
+            entity='account_type',
+	    #the response to prompt the user with if it is missing in the request 
+            responses=['Sure. for which account?'])
+        ],
+	 #keys to specify if you want to break out of the slot filling logic
+        'exit_keys' : ['cancel', 'restart', 'exit', 'reset', 'no', 'nevermind', 'stop', 'back', 'help', 'stop it', 'go back'
+                'new task', 'other', 'return'],
+	#a message to prompt the user after exit
+        'exit_msg' : 'A few other banking tasks you can try are, ordering checks and paying bills',
+	#the number of max tries for the user to specify the entity
+	'max_retries' = 1
+    }
+
+    #the @app.auto_fill decorator indicates it is a dialogue state handler that requires a form and uses the slot filling logic	
+    @app.auto_fill(intent='check_balances', form=balance_form)
+    #Control is passed on to this dialogue state handler one the slot-filling process is completed and all required entities in this form have been obtained.
+    def check_balances_handler(request, responder):
+        if not user_data: 
+            _pull_data(request)
+        if request.entities:
+            for entity in request.entities:
+                if entity['type'] == 'account_type':
+                    responder.slots['account'] = entity['value'][0]['cname'] or entity['text']
+                    responder.slots['amount'] = _get(entity['value'][0]['cname'] or entity['text'])
+                    responder.reply('Your {account} account balance is {amount}')
+        
+
+
+6. Knowledge Base
+^^^^^^^^^^^^^^^^^
+
+Since the banking assistant will require personal information we have decided to not include a knowledge base to mimic how this data may be received from an
+external API. Most likely the frontend would handle the user authentication and pass on a token to the request context, which is immutable once the conversation has started. The MindMeld server can then use this token to retrieve user information from a secure database. In this blueprint, we demonstrate how this can be done by simply passing the `user_name` in the context and using it to retrieve information from a simple database. For a production application, these can be replaced by a secure token and an API call to a secure database. These databases and APIs likely already exist internally at bank organizations.
```

Comment:
```suggestion
external API. Most likely the frontend would handle the user authentication and pass on a token to the request context, which is immutable once the conversation has started. The MindMeld server can then use this token to retrieve user information from a secure database. In this blueprint, we demonstrate how this can be done by simply passing the `user_name` in the context and using it to retrieve information from a simple database. For a production application, these can be replaced by a secure token and an API call to a secure database. These databases and APIs likely already exist internally at financial institutions.
```